skfda/misc/metrics/_sdtw_distances.py

```diff
@@ -0,0 +1,346 @@
+            "fdata1 and fdata2 must be FDataGrid objects"
+        )
+
+    # dim_codomain can be higher than one,
+    # but dim_domain must be one
+    if not fdata1.dim_domain == 1:
+        raise ValueError(
+            "fdata1 and fdata2 must have their domain dimension equal to one"
+        )
+
+    if not (fdata1.n_samples == 1 & fdata2.n_samples == 1):
+        raise ValueError(
+            "Both fdata1 and fdata2 must each contain a single sample"
+        )
+
+
+def half_sq_euclidean(
+    arg1: NDArrayFloat,
+    arg2: Optional[NDArrayFloat] = None,
+) -> NDArrayFloat:
+    """
+    Return the half squared Euclidean row distance between
+    two matrices of size (n1, d) and (n2, d).
+
+    Args:
+        arg1: First sample (n_samples_1, n_features).
+        arg2: Second sample (n_samples_2, n_features).
+            If it is ``None`` returns the half squared Euclidean distance
+            between the rows of ``arg1``.
+
+    The distance is taken w.r.t matrix rows.
+    """
+    if arg2 is not None:
+        # np.dot(X, Y.T)=inner_product_matrix(X, Y) but np ~1.6 faster
+        # X.shape is (n_X, d) and Y.shape is (n_Y, d)
+        cost_12 = -1 * inner_product_matrix(arg1, arg2)
+
+        # np.sum(X ** 2, axis=1)=inner_product(X, X)
+        cost_12 += (0.5 * inner_product(arg1, arg1))[:, np.newaxis]
+        cost_12 += 0.5 * inner_product(arg2, arg2)
+
+        return cost_12
+    else:
+        cost_11 = -1 * inner_product_matrix(arg1, arg2)
+        # half Sum Of Squares on each row
+        sos_row = 0.5 * inner_product(arg1, arg1)
+        cost_11 += sos_row[:, np.newaxis]
+        cost_11 += sos_row
+
+        return cost_11
+
+
+def _sdtw_divergence(
+    fdata1: FDataGrid,
+    fdata2: FDataGrid,
+    *,
+    gamma: float = 1.0,
+    cost: Union[Tuple[NDArrayFloat], Callable] = None,
+    check_cost: Boolean = False,
+) -> float:
+
+    # fdata1 and fdata2 are only required to be one-dimensional.
+    # They can have different (one-dimensional) domain ranges and
+    # different grid points size
+
+    _check_input_fdata(fdata1, fdata2)
+
+    n1, n2 = fdata1.grid_points[0].size, fdata2.grid_points[0].size
+
+    if gamma <= 0:
+        raise ValueError(
+            f"gamma was set to {gamma} but must be positive"
+        )
+
+    # cost must satisfy:shape=(n1, n2),nonnegativity,indiscernability
+    # and symmetry
+    if cost is not None:
+        if isinstance(cost, tuple):
+            if not len(cost) == 3:
+                raise ValueError(
+                    "When the alignment cost matrices are pre-computed, "
+                    "then 'cost' must be a list of three two-dimensional "
+                    "numpy arrays"
+                )
+            else:
+                # shape of cost_12, cost_11 and cost_22
+                expected_shapes = [(n1, n2), (n1, n1), (n2, n2)]
+                for idx, c in enumerate(cost):
+                    if not (isinstance(c, NDArrayFloat) and c.ndim == 2):
```

Comment:
I understand that you want to check every possible error and give an understandable message to the user, but for some things this is ridiculous: there is more checking that actual code!

I think that types should not be checked unless you need to execute different code for different types. Users could use mypy to check types. Also, the pythonic way is to not check for things that WILL fail if they are not right, such as the number of dimensions here. It IS important, however, to check for things that, if incorrect, would give nonsensical results, such as parameters that need to be in some range.

Reply:
Okay, I agree so I remove this format check.